src/ProjectOrigin.WalletSystem.IntegrationTests/TransferCertificateTests.cs

```diff
@@ -64,6 +64,49 @@ public async Task Transfer_SingleSlice_LocalWallet()
         await WaitForCertCount(certId, 4);
     }
 
+    [Fact]
+    public async Task Transfer_EntireCertificate_LocalWallet()
+    {
+        //Arrange
+        var client = new WalletService.WalletServiceClient(_grpcFixture.Channel);
+        var issuedAmount = 250u;
+        var transferredAmount = issuedAmount;
+
+        var (sender, senderHeader) = GenerateUserHeader();
+        var commitment = new SecretCommitmentInfo(issuedAmount);
+        var senderDepositEndpoint = await CreateWalletDepositEndpoint(sender);
+        var position = 1;
+        var issuedEvent = await _registryFixture.IssueCertificate(Electricity.V1.GranularCertificateType.Production, commitment, senderDepositEndpoint.PublicKey.Derive(position).GetPublicKey());
+        var certId = Guid.Parse(issuedEvent.CertificateId.StreamId.Value);
+        await InsertSlice(senderDepositEndpoint, position, issuedEvent, commitment);
+
+        var (_, recipientHeader) = GenerateUserHeader();
+        var createEndpointResponse = await client.CreateWalletDepositEndpointAsync(new CreateWalletDepositEndpointRequest(), recipientHeader);
+        var receiverEndpointResponse = await client.CreateReceiverDepositEndpointAsync(new CreateReceiverDepositEndpointRequest
+        {
+            WalletDepositEndpoint = createEndpointResponse.WalletDepositEndpoint
+        }, senderHeader);
+
+        //Act
+        var request = new TransferRequest()
+        {
+            CertificateId = issuedEvent.CertificateId,
+            Quantity = transferredAmount,
+            ReceiverId = receiverEndpointResponse.ReceiverId
+        };
+        await client.TransferCertificateAsync(request, senderHeader);
+
+        //Assert
+        await WaitForCertCount(client, recipientHeader, 1);
```

Comment:
I am not sure what this method does and why it is needed. Could the name be more specific?

Reply:
I have deleted this test and added a new one in the test of the certificates repository.